paasta_tools/kubernetes_tools.py

```diff
@@ -1322,6 +1324,32 @@ def get_pod_template_spec(
                 "termination_grace_period_seconds"
             ] = termination_grace_period
 
+        if self.get_iam_role_provider() == "aws":
+            annotations["iam.amazonaws.com/role"] = ""
```

Comment:
is this defensive against something else somewhere setting this attribute? in this case i would instead fail very loudly with something like `raise RuntimeError("can't have iam.amazonaws.com/role with aws provider")`. silently disappearing attributes can be quite annoying to debug.

Reply:
`iam.amazonaws.com/role` is used by `kiam`.
Previously we set `iam.amazonaws.com/role` to `""` if `am_role` wasn't set.
So, I just want to separate aws changes from kiam changes, and don't change anything related to kiam, to make sure that `kiam` will behave exactly the same way as before.
We could remove that later.